lib/reek/context/ghost_context.rb

```diff
@@ -0,0 +1,62 @@
+require_relative 'code_context'
+require_relative 'singleton_method_context'
+
+module Reek
+  module Context
+    # Semi-transparent context to represent a metaclass while building the
+    # context tree. This context will not be part of the resulting tree, but
+    # will track context and visibility separately while building is in
+    # progress.
+    class GhostContext
+      attr_reader :children
+      private_attr_reader :visibility_tracker
+
+      def initialize(parent, _exp)
+        @context = parent
```

Comment:
How about starting to correct one of the biggest naming errors in our code base and call this "@parent"? Or would you rather prefer to do this in a separate PR?